internal/provider/kubernetes/httproute.go

```diff
@@ -145,6 +147,20 @@ func (r *httpRouteReconciler) Reconcile(ctx context.Context, request reconcile.R
 		routeKey := utils.NamespacedName(&route)
 		if routeKey == request.NamespacedName {
 			found = true
+			gws, err := r.validateParentRefs(ctx, &route)
+			if err != nil {
+				// Remove the route from the watchable map since it's invalid.
+				r.resources.HTTPRoutes.Delete(routeKey)
+				return reconcile.Result{}, fmt.Errorf("invalid parentRefs for httproute %s/%s: %v",
```

Comment:
returning an error will re-enqueue the object for reconciliation again, I don't think we want to do that here, we can just re-reconcile if/when it's changed in the future.